### ISSUE TITLE
iOS - Crashlytics issue in flutter_inappwebview

### ISSUE THREAD

**Issue by marcqtan** [pichillilorenzo/flutter_inappwebview#789]
Firebase crashlytics is reporting a lot of crashes specifically in flutter_inappwebview for IOS (v13 upto v14.4.2) but I cant reproduce the crash in my device. Also 600users are affected with this crash.

Here's the stacktrace: 

```
Crashed: com.apple.main-thread
0  flutter_inappwebview           0x1043e07a0 $s20flutter_inappwebview12InAppWebViewC03webF0_15decidePolicyFor15decisionHandlerySo05WKWebF0C_So18WKNavigationActionCySo0noI0VctF06$sSo24noI16VIeyBy_ABIegy_TRALIeyBy_Tf1nncn_nTf4dnng_n + 972
1  flutter_inappwebview           0x1043cb5d8 $s20flutter_inappwebview12InAppWebViewC03webF0_15decidePolicyFor15decisionHandlerySo05WKWebF0C_So18WKNavigationActionCySo0noI0VctFToTm + 100
2  WebKit                         0x196488f14 WebKit::NavigationState::NavigationClient::decidePolicyForNavigationAction(WebKit::WebPageProxy&, WTF::Ref<API::NavigationAction, WTF::DumbPtrTraits<API::NavigationAction> >&&, WTF::Ref<WebKit::WebFramePolicyListenerProxy, WTF::DumbPtrTraits<WebKit::WebFramePolicyListenerProxy> >&&, API::Object*) + 1168
3  WebKit                         0x19652ffa8 WebKit::WebPageProxy::decidePolicyForNavigationAction(WTF::Ref<WebKit::WebProcessProxy, WTF::DumbPtrTraits<WebKit::WebProcessProxy> >&&, WebKit::WebFrameProxy&, WebKit::FrameInfoData&&, unsigned long long, WebKit::NavigationActionData&&, WebKit::FrameInfoData&&, WTF::Optional<WTF::ObjectIdentifier<WebKit::WebPageProxyIdentifierType> >, WebCore::ResourceRequest const&, WebCore::ResourceRequest&&, IPC::FormDataReference&&, WebCore::ResourceResponse&&, WebKit::UserData const&, WTF::Ref<WebKit::WebPageProxy::PolicyDecisionSender, WTF::DumbPtrTraits<WebKit::WebPageProxy::PolicyDecisionSender> >&&) + 3732
4  WebKit                         0x19652f06c WebKit::WebPageProxy::decidePolicyForNavigationActionAsyncShared(WTF::Ref<WebKit::WebProcessProxy, WTF::DumbPtrTraits<WebKit::WebProcessProxy> >&&, WTF::ObjectIdentifier<WebCore::PageIdentifierType>, WTF::ObjectIdentifier<WebCore::FrameIdentifierType>, WebKit::FrameInfoData&&, WebCore::PolicyCheckIdentifier, unsigned long long, WebKit::NavigationActionData&&, WebKit::FrameInfoData&&, WTF::Optional<WTF::ObjectIdentifier<WebKit::WebPageProxyIdentifierType> >, WebCore::ResourceRequest const&, WebCore::ResourceRequest&&, IPC::FormDataReference&&, WebCore::ResourceResponse&&, WebKit::UserData const&, unsigned long long) + 356
5  WebKit                         0x1964be224 WebKit::ProvisionalPageProxy::decidePolicyForNavigationActionAsync(WTF::ObjectIdentifier<WebCore::FrameIdentifierType>, WebKit::FrameInfoData&&, WebCore::PolicyCheckIdentifier, unsigned long long, WebKit::NavigationActionData&&, WebKit::FrameInfoData&&, WTF::Optional<WTF::ObjectIdentifier<WebKit::WebPageProxyIdentifierType> >, WebCore::ResourceRequest const&, WebCore::ResourceRequest&&, IPC::FormDataReference&&, WebCore::ResourceResponse&&, WebKit::UserData const&, unsigned long long) + 188
6  WebKit                         0x1964bef8c WebKit::ProvisionalPageProxy::didReceiveMessage(IPC::Connection&, IPC::Decoder&) + 2012
7  WebKit                         0x196224a1c IPC::MessageReceiverMap::dispatchMessage(IPC::Connection&, IPC::Decoder&) + 124
8  WebKit                         0x1965595dc WebKit::WebProcessProxy::didReceiveMessage(IPC::Connection&, IPC::Decoder&) + 40
9  WebKit                         0x196207c4c IPC::Connection::dispatchMessage(std::__1::unique_ptr<IPC::Decoder, std::__1::default_delete<IPC::Decoder> >) + 852
10 WebKit                         0x196207490 IPC::Connection::dispatchIncomingMessages() + 460
11 JavaScriptCore                 0x193f305e0 WTF::RunLoop::performWork() + 484
12 JavaScriptCore                 0x193f312c8 WTF::RunLoop::performWork(void*) + 36
13 CoreFoundation                 0x189bc176c __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
14 CoreFoundation                 0x189bc1668 __CFRunLoopDoSource0 + 208
15 CoreFoundation                 0x189bc0960 __CFRunLoopDoSources0 + 268
16 CoreFoundation                 0x189bbaa8c __CFRunLoopRun + 824
17 CoreFoundation                 0x189bba21c CFRunLoopRunSpecific + 600
18 GraphicsServices               0x1a1784784 GSEventRunModal + 164
19 UIKitCore                      0x18c5faee8 -[UIApplication _run] + 1072
20 UIKitCore                      0x18c60075c UIApplicationMain + 168
21 Runner                         0x1023cbb60 main + 6 (AppDelegate.swift:6)
22 libdyld.dylib                  0x18987a6b0 start + 4
```

Image from firebase:

<img width="1269" alt="Screen Shot 2021-04-07 at 5 07 04 PM" src="https://user-images.githubusercontent.com/66229352/113840901-c38ac300-97c3-11eb-907e-2be31ead2d13.png">

Can you please advise? Thank you!

**Comment by pichillilorenzo** [comment]
Please share more context, for example, the plugin version used.
Also, could you post the code related to my plugin or a similar code? Maybe there is something that you are doing wrong or using in the wrong way.
Unfortunately, there is no specific code line from your logs to my plugin code in order to have a better understanding of it.

**Comment by pichillilorenzo** [comment]
Also, could you post the full crash message log, please?

**Comment by pichillilorenzo** [comment]
Read [Adding Identifiable Symbol Names to a Crash Report](https://developer.apple.com/documentation/xcode/diagnosing_issues_using_crash_reports_and_device_logs/adding_identifiable_symbol_names_to_a_crash_report) to give me a better message log

**Comment by pichillilorenzo** [comment]
Read also [Diagnosing Issues Using Crash Reports and Device Logs](https://developer.apple.com/documentation/xcode/diagnosing_issues_using_crash_reports_and_device_logs)

**Comment by marcqtan** [comment]
I've finally managed to reproduced this issue. It's only happening in IOS.
Below is an example code to reproduced it

Steps:
1. Click the link
2. Click the Appstore or the Playstore Image
3. Crash will happen.

It seems it will crash because of accessing target="_blank" twice? 

I'm using flutter_inappwebview: ^4.0.0+4. I can't use version 5 since the project is really old and at the moment, there's no plan to update it. Can you help me fix this and only using version 4 of inappwebview? 

Thanks!

Code:
```
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:flutter_icons/flutter_icons.dart';
import 'package:flutter_inappwebview/flutter_inappwebview.dart';

class DownloadCenterTab extends StatefulWidget {
  @override
  _DownloadCenterTabState createState() => _DownloadCenterTabState();
}

class _DownloadCenterTabState extends State<DownloadCenterTab> {
  double _progress = 0;
  InAppWebViewController _controller;

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: Scaffold(
        resizeToAvoidBottomInset: false,
        body: Column(
          children: [
            Expanded(
              child: Container(
                child: Stack(
                  children: [
                    InAppWebView(
                      initialOptions: InAppWebViewGroupOptions(
                        crossPlatform: InAppWebViewOptions(
                          useOnDownloadStart: true,
                        ),
                      ),
                 
                      initialData: InAppWebViewInitialData(
                        data: '''<!DOCTYPE html>
                                <html>
                                <body>
                                <a href="https://www.booco.jp/" target="_blank">英語学習 booco</a>
                                </body>
                                </html>''',
                      ),
                      onProgressChanged: (_, progress) {
                        setState(() {
                          _progress = progress / 100;
                        });
                      },

                      onWebViewCreated: (controller) {
                        _controller = controller;
                      },
                      onCreateWindow:
                          (controller, onCreateWindowRequest) async {
                        await controller.loadUrl(
                          url: onCreateWindowRequest.url,
                        );

                        return true;
                      },
                    ),
                    Center(
                      child: _progress < 1
                          ? CircularProgressIndicator()
                          : Container(),
                    ),
                  ],
                ),
              ),
            ),
            Ink(
              height: kToolbarHeight,
              color: Theme.of(context).primaryColor,
              child: ButtonBar(
                children: [
                  IconButton(
                    color: Colors.white,
                    icon: Icon(
                      SimpleLineIcons.arrow_left,
                    ),
                    onPressed: () async {
                      await _controller.goBack();
                    },
                  ),
                  IconButton(
                    color: Colors.white,
                    icon: Icon(
                      SimpleLineIcons.refresh,
                    ),
                    onPressed: () {
                      _controller.reload();
                    },
                  ),
                  IconButton(
                    color: Colors.white,
                    icon: Icon(
                      SimpleLineIcons.arrow_right,
                    ),
                    onPressed: () {
                      _controller.goForward();
                    },
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}

```

**Comment by pichillilorenzo** [comment]
Unfortunately, this has been fixed with version 5.
Version 4 is not supported anymore and you should use the latest one.
You can use the latest version and use the `--no-sound-null-safety` flag when you execute your Flutter App. Check https://dart.dev/null-safety/unsound-null-safety#testing-or-running-mixed-version-programs for more details.

**Comment by marcqtan** [comment]
Thank you for answering. I've tried upgrading to version 5 and use flutter run --no-sound-null-safety, but I got some errors during build.

```
../../flutter/.pub-cache/hosted/pub.dartlang.org/extended_image-0.9.0/lib/src/extended_image.dart:743:44: Error: No named parameter with the name 'nullOk'.
    _invertColors = MediaQuery.of(context, nullOk: true)?.invertColors ??
```

It seems that I still need to update all project to support Flutter 2 nullsafety even if i'm already using --no-sound-null-safety  flag.. So it's a no go.

As I've mentioned earlier, Is there a way to apply the fix and still using version 4? 
I can cherry-picked it one by one in my forked inappwebview.  But I don't know which changes should I cherry-pick. Is it a big change?
